README.md

```diff
@@ -82,66 +76,87 @@ The `samlconf` script may take the following parameters:
-                If it is not given, this test kit will only verify that a valid SAML error 
+                If it is not given, this test kit will only verify that a valid SAML error
                 response is returned.
 
            -u username:password
                 The username and password to use when logging in.
                 The default value is admin:admin.
 
-## Project Structure
-This section will briefly talk about the project structure.
-
-### ctk
-This module contains all the test related modules.
-
-#### idp
-This module will contain all tests being written against a SAML IdP. The `src` directory of the module is organized by the SAML specification as follows:
+## Developing
 
-* Package: Based on Profile (i.e. WebSSO, Single Logout)
-  * Class: Based on Binding (i.e. POST, REDIRECT, ARTIFACT)
+#### Developing a plugin
 
-The [coverage](ctk/idp/coverage) directory is used to track which sections of each SAML specification are covered by these tests. It also includes a list of MUSTs that are not tested and justification for each.
+Before testing an IdP, it has to be connected with the SAML-CTK. To do so, a custom plugin is required.
 
-#### common
-This module contains all the classes relating to utility for and verification of the test classes.
+###### Why is a plugin required?
 
-> NOTE 
->
-> Schema verification is run before other tests and verifications.
+When logging in to an IdP using the Single Sign-On protocol, there are 6 steps:
 
-### library
-This module contains an assortment of Java classes that have been copied over from DDF to support operations that shouldn't be handled by the test code; i.e. signature validation using x509 certificates.
+1. HTTP Request to Service Provider
+1. Service Provider Determines Identity Provider
+1. `<AuthnRequest>` issued by Service Provider to Identity Provider
+1. Identity Provider identifies Principal
+1. Identity Provider issues `<Response>` to Service Provider
+1. Service Provider grants or denies access to Principal
 
-### external
-This module contains anything related to a specific SAML implementing product.
+The specification mentions that in step 4, "the principal is identified by the identity provider by some means outside the scope of \[the Single Sign-On\] profile".
+Since the authentication step is implementation specific, each IdP must write a plugin which will process the authentication of a principal.
+See [4.1 Web Browser SSO Profile](https://www.oasis-open.org/committees/download.php/56782/sstc-saml-profiles-errata-2.0-wd-07.pdf) for more information on the different steps for the SSO profile.
 
-#### samlconf-plugins-api
-This module contains the API that must be implemented for a SAML product in order to run this test kit against that product.
+###### How to implement a plugin jar?
 
-#### implementations
-This module contains implementations of the API for specific SAML products.
+1. Implement a plugin jar for the IdP's authentication implementation. \
```

Comment:
✏️ Should probably mention here that the implementation class is implementing a Java service and that either a hand-crafted services reference file (`META-INF/services/org.codice.compliance.saml.plugin.IdpSSOResponder`) will need to be added to their jar or the implementors will want to annotate their implementation with `@MetaInfServices` and use the kapt plugin.